codeforamerica/vita-min
# Make state file FAQ page show only the previous active season's states until the start of next season

## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-442
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- The story suggests doing a feature flag but I think this will be a recurring issue if the product exists for multiple years so I added a property to the state information service to track which years states are/were active. I think this can probably be used elsewhere too.
- The code in the controller index is slightly confusing imo but i couldn't think of a better way to handle the cases. I guess we could assume that the start date will always be on Jan 1 and in that case we'd just use the current tax year every time. Open to thoughts!
## How to test?
- I added a controller spec that tests essentially 2 cases: 1) when it's before the start of the season in the year *of* the season and 2) when it's past the start date. I couldn't think of any other cases that seemed important but I might be missing something!
- Manual testing: go to the FAQ page and see that NC disappears
  - I'd love to be able to test the case where NC is open but the config doesn't have next year's start date so I was only able to test it locally by changing the development config. It's kind of the same test though.
## Screenshots (for visual changes)
- Before
![image](https://github.com/user-attachments/assets/ba2b8416-322d-4fa5-8224-089be255adfe)
- After
![image](https://github.com/user-attachments/assets/304dffe5-e3d6-434e-a2a8-c4d2a0e1cd09)
